dashboard-ui/src/pages/home.test.tsx

```diff
@@ -32,3 +42,91 @@ describe('home page', () => {
+  it('returns undefined when fetching is true', () => {
+    const result = applySearchAndFilter(true, workloadItemMock, '', '');
+
+    expect(result).toBeUndefined();
+  });
+
+  it('returns undefined if the workload item is undefined or null', () => {
+    const firstResult = applySearchAndFilter(false, undefined, '', '');
+    const secondResult = applySearchAndFilter(false, null, '', '');
+
+    expect(firstResult).toBeUndefined();
+    expect(secondResult).toBeUndefined();
+  });
+
+  it('returns only the non deleted workload items', () => {
+    const result = applySearchAndFilter(false, workloadItemMock, '', '');
+    const nonDeletedWorkload = workloadItemMock.filter((item) => item.metadata.deletionTimestamp === null);
+
+    expect(result).toEqual(nonDeletedWorkload);
+  });
+
+  it('returns workload items within selected namespace and search string', () => {
+    const namespace = 'kube-system';
+    const search = 'kind';
+
+    const result = applySearchAndFilter(false, workloadItemMock, search, namespace);
+    const output = workloadItemMock.filter((item) => item.metadata.namespace === namespace && item.metadata.name.includes(search));
+
+    expect(result).toEqual(output);
+  });
+
+  it('returns all workloads from selected namespace', () => {
+    const firstResult = applySearchAndFilter(false, workloadItemMock, '', 'kube-system');
+    const output = workloadItemMock.filter((item) => item.metadata.namespace === 'kube-system');
+    expect(firstResult).toEqual(output);
+
+    // returns empty array as the only workload item in namespace kubetail-system has a deletion timestamp
+    const secondResult = applySearchAndFilter(false, workloadItemMock, '', 'kubetail-system');
+    expect(secondResult).toEqual([]);
+  });
+});
+
+describe('noSearchResults', () => {
+  it('returns true if all workload items are undefined', () => {
+    const result = noSearchResults(undefined, undefined, undefined);
+    expect(result).toEqual(true);
+  });
+
+  it('returns true if all workload items are either undefined or empty arrays', () => {
+    const result = noSearchResults(undefined, undefined, undefined, [], []);
+    expect(result).toEqual(true);
+  });
+
+  it('returns false if any or every workload item has alteast one item in it', () => {
+    const firstResult = noSearchResults(workloadItemMock, workloadItemMock, workloadItemMock, workloadItemMock);
+    const secondResult = noSearchResults(workloadItemMock, workloadItemMock, undefined, []);
+
+    expect(firstResult).toEqual(false);
+    expect(secondResult).toEqual(false);
+  });
+});
+
+describe('getContainerIDs', () => {
+  it('returns an empty array if parentId is not present in the ownership map', () => {
+    const result = getContainerIDs('dc8fbace-67c0-aaff6dbe2d7a-id-not-present', ownershipMapMock);
+
+    expect(result).toEqual([]);
+  });
+
+  it('returns all the values of a parentId when none of the childIds have grandChildIds', () => {
```

Comment:
"returns all the values of a parentId" is a bit confusing. maybe something more like "returns immediate children of a parent"...